### PR TITLE
Margin

### PR DESCRIPTION
**Pull request by bamooxa**
Better core formula.
I've been running for some months now and I haven't come across the instant/unnecessary trade bug (A>B>C instantly).
Every trade results in atleast my scout margin percentage coin profit.

### CODE REVIEW

**Comment by edeng23** [comment]
I like this change, it makes more sense than my original formula. Will run a few tests and hopefully merge soon.

File: binance_trade_bot/auto_trader.py [hunk 1]
```diff
@@ -123,13 +123,19 @@ def _get_ratios(self, coin: Coin, coin_price):
...
-                coin_opt_coin_ratio - transaction_fee * self.config.SCOUT_MULTIPLIER * coin_opt_coin_ratio
-            ) - pair.ratio
+            # Fees
+            from_fee = self.manager.get_fee(pair.from_coin, self.config.BRIDGE, True)
+            to_fee = self.manager.get_fee(pair.to_coin, self.config.BRIDGE, False)
+            transaction_fee = from_fee + to_fee - from_fee * to_fee
+
+            if self.config.USE_MARGIN == "yes":
+                ratio_dict[pair] = (
+                    (1- transaction_fee) * coin_opt_coin_ratio / pair.ratio - 1 - self.CONFIG.SCOUT_MARGIN / 100
```
Comment (yigitakbulut) [inline binance_trade_bot/auto_trader.py]: Here it should be self.**config**.SCOUT_MARGIN

File: binance_trade_bot/config.py [hunk 2]
```diff
@@ -70,3 +72,6 @@ def __init__(self):
 
         self.SELL_TIMEOUT = os.environ.get("SELL_TIMEOUT") or config.get(USER_CFG_SECTION, "sell_timeout")
         self.BUY_TIMEOUT = os.environ.get("BUY_TIMEOUT") or config.get(USER_CFG_SECTION, "buy_timeout")
+
+        self.USE_MARGIN = os.environ.get("USE_MARGIN") or config.get(USER_CFG_SECTION, "use_margin")
+        self.SCOUT_MARGIN = os.environ.get("SCOUT_MARGIN") or config.get(USER_CFG_SECTION, "scout_margin")
```
Comment (yigitakbulut) [inline binance_trade_bot/config.py]: self.SCOUT_MARGIN = **float(** os.environ.get("SCOUT_MARGIN") or config.get(USER_CFG_SECTION, "scout_margin")**)**

**Review by yigitakbulut — CHANGES_REQUESTED** [review]
Very good! Thank you for this, I think the strategy works better with this.
There is a typo in auto_trader.py, also you should convert the scout margin to float when you import it from config.

**Review by yigitakbulut — APPROVED** [review]

**Comment by lisabbasil** [comment]
I'm a bit confused about the fees. They are given as a percentage, so should be multiplied with the volume that will be traded, not with a ratio, or a ratio of ratios. Guess I just don't understand the formula?

**Comment by rogerthomas84** [comment]
I really like this proposed change 👍

**Comment by muhammadnumair** [comment]
Thanks for the change, @edeng23 were you able to test it?

**Pull request merged**